tests/parsers/beckman_pharmspec/beckman_pharmspec_parser_test.py

```diff
@@ -22,129 +22,223 @@
 
 
 @pytest.fixture()
-def test_file() -> Path:
+def test_file_1() -> Path:
+    """Test file 1 has two runs and an average distribution.
+
+    :return: Path to test file 1
+    """
     f = Path(__file__).parent / "testdata/hiac_example_1.xlsx"
     return f.absolute()
 
 
+@pytest.fixture()
+def test_file_2() -> Path:
+    """Test file 2 has 3 runs and an average distribution.
+
+    :return: Path to test file 2
+    """
+    f = Path(__file__).parent / "testdata/hiac_example_2.xlsx"
+    return f.absolute()
+
+
+@pytest.fixture()
+def test_file_3() -> Path:
+    """Test file 3 has 1 run and an average distribution.
+
+    :return: Path to test file 3
+    """
+    f = Path(__file__).parent / "testdata/hiac_example_3.xlsx"
+    return f.absolute()
+
+
+@pytest.fixture()
+def test_file_4() -> Path:
+    """Test file 4 has 1 run and no calculated distribution.
+
+    :return: Path to test file 4
+    """
+    f = Path(__file__).parent / "testdata/hiac_example_4.xlsx"
+    return f.absolute()
+
+
+@pytest.fixture()
+def test_file_5() -> Path:
+    """Test file 5 has 3 runs and no calculated distribution.
+
+    :return: Path to test file 5
+    """
+    f = Path(__file__).parent / "testdata/hiac_example_5.xlsx"
+    return f.absolute()
+
+
+@pytest.fixture()
+def test_files(
+    test_file_1: Path,
+    test_file_2: Path,
+    test_file_3: Path,
+    test_file_4: Path,
+    test_file_5: Path,
+) -> dict[str, Path]:
+    return {
+        "test_file_1": test_file_1,
+        "test_file_2": test_file_2,
+        "test_file_3": test_file_3,
+        "test_file_4": test_file_4,
+        "test_file_5": test_file_5,
+    }
+
+
 @pytest.mark.short
-def test_get_model(test_file: Path) -> None:
-    parser = PharmSpecParser(TimestampParser())
+def test_get_model(test_files: dict[str, Path]) -> None:
```

Comment:
As part of the request to change how we are passing test files, you can use parametrize to create test cases. `unchained_labs_lunatic_structure_test` has an example of multi parameters.

Something like:

```
@pytest.mark.parametrize(
    "file_name,num_measurement_docs,has_calculated_doc",
    [
        (OUTPUT_FILES[0], 2, True),
        ...
     ]
)

Reply:
Fixed